### PR TITLE
feat(run_out_module): add tests to run out

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description
Add tests to increase coverage.

image

Note that methods that heavily depend on ROS2 are skipped for these tests/evaluations.
## Related links

**Parent Issue:**

- Link

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_dynamic_obstacle.cpp [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+TEST_F(TestDynamicObstacleMethods, testConcatPointCloud)
+{
+  constexpr size_t n_points{10};
+  constexpr double points_resolution{0.025};
+
+  pcl::PointCloud<pcl::PointXYZ> point_cloud_1 = generate_pointcloud(n_points, points_resolution);
+  pcl::PointCloud<pcl::PointXYZ> point_cloud_2 =
+    generate_pointcloud(n_points * 2, points_resolution * 2.0);
+  auto point_cloud_concat = concatPointCloud(point_cloud_1, point_cloud_2);
+  EXPECT_TRUE(point_cloud_concat.data.size() >= point_cloud_1.size() + point_cloud_2.size());
```
Comment (kosuke55) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_dynamic_obstacle.cpp:321]: I did not check the details but not exact == ?

File: planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_path_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  const auto geom_p_med = autoware::universe_utils::createPoint(
+    p_med.point.pose.position.x, p_med.point.pose.position.y, p_med.point.pose.position.z);
+  std::vector<Point> dst_points{geom_p_src, geom_p_dst, geom_p_med};
+  const auto closest_point_src = findLongitudinalNearestPoint(path.points, geom_p_src, dst_points);
+  const auto closest_point_dst = findLongitudinalNearestPoint(path.points, geom_p_dst, dst_points);
+  const auto closest_point_med = findLongitudinalNearestPoint(path.points, geom_p_med, dst_points);
+
+  EXPECT_DOUBLE_EQ(
+    autoware::universe_utils::calcDistance3d(closest_point_src, geom_p_src),
+    autoware::universe_utils::calcDistance3d(geom_p_src, geom_p_src));
```
Comment (kosuke55) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_path_utils.cpp:64]: same to
```suggestion
  EXPECT_TRUE(
    autoware::universe_utils::calcDistance3d(closest_point_src, geom_p_src) < std::numeric_limits<float>::epsilon());)
```
?

File: planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_dynamic_obstacle.cpp [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+};
+namespace autoware::behavior_velocity_planner
+{
+TEST_F(TestDynamicObstacleMethods, testCreateQuaternionFacingToTrajectory)
+{
+  constexpr size_t n_path_points{10};
+  PathPointsWithLaneId path;
+  PathPointWithLaneId base_point;
+  for (size_t i = 0; i < n_path_points; ++i) {
+    const PathPointWithLaneId p = createExtendPathPoint(static_cast<double>(i), base_point);
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_dynamic_obstacle.cpp:124]: @kosuke55 In this case I want to use the createExtendPathPoint function of the run out module. I believe that function internally calls autoware::universe_utils::calcOffsetPose

File: planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_path_utils.cpp (hunk 2 above)
Comment (danielsanchezaran) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_path_utils.cpp:64]: @kosuke55 that should work too, but I just wanted to keep it consistent with the other checks 
`  EXPECT_DOUBLE_EQ(
    autoware::universe_utils::calcDistance3d(closest_point_dst, geom_p_dst),
    autoware::universe_utils::calcDistance3d(geom_p_src, geom_p_dst));
  EXPECT_DOUBLE_EQ(
    autoware::universe_utils::calcDistance3d(closest_point_med, geom_p_med),
    autoware::universe_utils::calcDistance3d(geom_p_src, geom_p_med));`

Comment (danielsanchezaran) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_run_out_module/tests/test_path_utils.cpp:64]: @kosuke55 Those points are actually not the same, so your suggested change does not work when evaluating the function. The evaluated method is supposed to return the point with the smallest signed longitudinal distance, so it always returns the the one point that is the most "behind" the path. I put a comment on the function to clarify that.

**Review by kosuke55 — APPROVED** [review]

**Pull request merged**